core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/cloudmapper/CloudMapperSourceOpDesc.scala

```diff
@@ -24,7 +27,33 @@ class CloudMapperSourceOpDesc extends PythonSourceOperatorDescriptor {
   @JsonPropertyDescription("Cluster Id")
   val clusterId: String = ""
 
+  @JsonIgnore
+  var filePath: Option[String] = None
+
+  @JsonIgnore
+  var datasetFile: Option[DatasetFileDocument] = None
+
+  def determineFilePathOrDatasetFile(fileName: Option[String]): (String, DatasetFileDocument) = {
```

Comment:
Since the function `determineFilePathOrDatasetFile` is a duplicate of the method in the `ScanSourceOpDesc` class, it would be better to separate this function out of the `ScanSourceOpDesc`. 

I think you can move the function `determineFilePathOrDatasetFile`  to `core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/util/` , creating a new object called `OperatorFilePathUtils.scala`, and put the function in here.